### ISSUE TITLE
Build failed on `ubuntu 20.04` and `Arrow 5.0.0`

### ISSUE THREAD

**Issue by mjsML** [pbotros/river#7]
Hello,
I tried to install river, I installed arrow as per the instructions for ubuntu [here](https://arrow.apache.org/install/) (the `C++ and GLib (C) Packages for Debian GNU/Linux, Ubuntu, CentOS and Amazon Linux` section).

But when I build with the `-DRIVER_BUILD_INGESTER=1`  I get :
```
In file included from /home/mjs/river/ingester/src/ingester_test_utils.h:13,
                 from /home/mjs/river/ingester/src/integration_test.cpp:9:
/home/mjs/river/ingester/src/integration_test.cpp: In member function ‘void IntegrationTest::run()’:
/home/mjs/river/ingester/src/integration_test.cpp:150:115: error: no matching function for call to ‘arrow::Array::View(std::shared_ptr<arrow::DataType>, std::shared_ptr<arrow::Array>*)’
  150 | (), reinterpret_cast<std::shared_ptr<arrow::Array> *>(&array)));
      |                                                              ^

In file included from /usr/include/arrow/array.h:25,
                 from /usr/include/arrow/api.h:22,
                 from /home/mjs/river/ingester/src/ingester_test_utils.h:8,
                 from /home/mjs/river/ingester/src/integration_test.cpp:9:
/usr/include/arrow/array/array_base.h:144:34: note: candidate: ‘arrow::Result<std::shared_ptr<arrow::Array> > arrow::Array::View(const std::shared_ptr<arrow::DataType>&) const’
  144 |   Result<std::shared_ptr<Array>> View(const std::shared_ptr<DataType>& type) const;
      |                                  ^~~~
/usr/include/arrow/array/array_base.h:144:34: note:   candidate expects 1 argument, 2 provided
make[2]: *** [ingester/src/CMakeFiles/ingester_test.dir/build.make:118: ingester/src/CMakeFiles/ingester_test.dir/integration_test.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3243: ingester/src/CMakeFiles/ingester_test.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

```

Is river compatible with the latest arrow libs? or I installed a wrong library/version?
Thanks in advance.

**Comment by pbotros** [comment]
Thanks @mjsML - indeed I've only tested it with Apache Arrow 0.16.0 so far. Looks like there was a breaking API change in the recent releases. I'll take a look to see how to handle the new Arrow.

**Comment by pbotros** [comment]
@mjsML fix https://github.com/pbotros/river/pull/10 merged that seems to fix it on my computer (Mac OSX) and tests still pass. Let me know if latest master fixes this for you, thanks!

**Comment by mjsML** [comment]
#10 indeed fixes the issue on ubuntu 20.04 ... Thank you!
